test/common/index.js

```diff
@@ -519,6 +519,49 @@ function mustNotCall(msg) {
   };
 }
 
+const _canNotMutateObjectDeepProxies = new WeakMap();
+
+function canNotMutateObjectDeep(original) {
+  if (original === null || typeof original !== 'object') {
```

Comment:
It'd be nice to have comments to clarify that functions are ignored on purpose. JSDoc could be nice also if you feel like it. Non-blocking comment.

Reply:
Added a comment via https://github.com/nodejs/node/pull/43196/commits/aa57ef751d9649f149e12115136133da332a8a9b. No feelings towards JSDoc since it's not really used in that file. :)